### PR TITLE
Features/download datapackage 305

### PR DESCRIPTION
**Pull request by MGlauer**
Add download for datapackages. 

* Add `form=datapackage` option to simple API
  * Returns  `{schema}_{ŧable}.zip` containing
    * `data.csv`: result of `form=csv"
    * `datapackage.json`: table's metadata in JSON-format
* Add a button for datapackage-download

### CODE REVIEW

**Comment by MGlauer** [comment]
Closes #305

**Comment by christian-rli** [comment]
I tried to have a look at the button and its functionality, but in my local instance of the oep I somehow cannot open the dataedit view. The error states that authentication for user none has failed (I tried logged in and not loged in). Full error message:


```
Request Method: | GET
-- | --
http://127.0.0.1:8000/dataedit/schemas
3.1
OperationalError
(psycopg2.OperationalError) FATAL:  Passwort-Authentifizierung für Benutzer »None« fehlgeschlagen FATAL:  Passwort-Authentifizierung für Benutzer »None« fehlgeschlagen  (Background on this error at: http://sqlalche.me/e/13/e3q8)
/home/christianh/anaconda3/lib/python3.7/site-packages/psycopg2/__init__.py, line 130, in connect
/home/christianh/anaconda3/bin/python
3.7.1
['/home/christianh/lokal/OpenEnergyFamily/OEP aufsetzen/oep-website',  '/home/christianh/anaconda3/lib/python37.zip',  '/home/christianh/anaconda3/lib/python3.7',  '/home/christianh/anaconda3/lib/python3.7/lib-dynload',  '/home/christianh/anaconda3/lib/python3.7/site-packages',  '/home/christianh/anaconda3/lib/python3.7/site-packages/metadata_tool-0.0.0-py3.7.egg',  '/home/christianh/anaconda3/lib/python3.7/site-packages/rdflib-4.2.2-py3.7.egg',  '/home/christianh/anaconda3/lib/python3.7/site-packages/isodate-0.6.0-py3.7.egg',  '/home/christianh/anaconda3/lib/python3.7/site-packages/feedin_germany-0.0.1-py3.7.egg',  '/home/christianh/anaconda3/lib/python3.7/site-packages/feedinlib-0.0.12-py3.7.egg',  '/home/christianh/anaconda3/lib/python3.7/site-packages/windpowerlib-0.0.4-py3.7.egg',  '/home/christianh/anaconda3/lib/python3.7/site-packages/pvlib-0.6.3-py3.7.egg']
Mon, 17 Aug 2020 18:19:21 +0200
```

Somehow that error rings familiar. Are you able to help with a quick long distance troubleshoot?

**Comment by christian-rli** [comment]
I fixed my setup. It had to do with my (old) database settings.

File: api/views.py [hunk 1]
```diff
@@ -631,7 +631,33 @@ def get(self, request, schema, table, row_id=None):
...
+            csv_name = "{schema}__{table}.csv".format(
+                schema=schema, table=table
+            )
+            zf.write_iter(csv_name, (
+                writer.writerow(x).encode("utf-8")
+                for x in itertools.chain([cols], return_obj["data"])
+            ))
+            table_obj = actions._get_table(schema=schema, table=table)
+            if table_obj.comment:
+                zf.writestr("metadata.json", table_obj.comment.encode("utf-8"))
```
Comment (christian-rli) [inline api/views.py:647]:
```suggestion
                zf.writestr("datapackage.json", table_obj.comment.encode("utf-8"))
```

File: api/views.py [hunk 2]
```diff
@@ -631,7 +631,33 @@ def get(self, request, schema, table, row_id=None):
...
+            else:
+                zf.writestr("metadata.json", json.dumps(JSONCompiler().visit(OEPMetadata())).encode("utf-8"))
```
Comment (christian-rli) [inline api/views.py:649]:
```suggestion
                zf.writestr("datapackage.json", json.dumps(JSONCompiler().visit(OEPMetadata())).encode("utf-8"))
```

**Review by christian-rli — CHANGES_REQUESTED** [review]
Downloads work. All expected files are there. I had a look at the [datapackage specification](https://frictionlessdata.io/data-package/#examples) and apparently the metadata file should be named datapackage.json . I made suggestions to the code. I think that's all that's needed. Once these changes are done i think it's good to go!

**Review by christian-rli — APPROVED** [review]
I tested the branch with my changes and the (renamed) metadatafile shows up as expected. It's ready to be merged from my perspective.

**Pull request merged**